core-backend/wasmi/src/funcs/mod.rs

```diff
@@ -1197,6 +1198,63 @@ where
         Func::wrap(store, func)
     }
 
+    pub fn random(store: &mut Store<HostState<E>>, forbidden: bool, memory: WasmiMemory) -> Func {
+        let func = move |mut caller: wasmi::Caller<'_, HostState<E>>,
+                         subject_ptr: u32,
+                         subject_len: u32,
+                         random_ptr: u32,
+                         bn_ptr: u32|
+              -> EmptyOutput {
+            exit_if!(forbidden, caller);
```

Comment:
please rebase on master and update this calls: updating global counters should be included